client/src/pages/admin/merge/MergePeople.js

```diff
@@ -75,10 +76,15 @@ const MergePeople = ({ pageDispatchers }) => {
   const person2 = mergeState[MERGE_SIDES.RIGHT]
   const mergedPerson = mergeState.merged
 
+  const nonMatchingRoles = person1 && person2 && person1.role !== person2.role
+
   return (
     <Container fluid>
       <Row>
-        <Messages error={saveError} />
+        <Messages
+          error={saveError}
+          warning={nonMatchingRoles ? NONMATCHING_ROLE_WARNING : ""}
```

Comment:
This is the only time you check for `nonMatchingRoles`, the other 15(!) times it is `!nonMatchingRoles`; maybe invert this to `matchingRoles`?

Reply:
Very good suggestion